src/components/services/Rtorrent.vue

```diff
@@ -0,0 +1,158 @@
+            </span>
+        </template>
+      </p>
+    </template>
+    <template #indicator>
+      <span v-if="!error" class="count">{{ count }}
+        <template v-if="count === 1">torrent</template>
+        <template v-else>torrents</template>
+      </span>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import Generic from './Generic.vue';
+
+// Units to add to download and upload rates.
+const units = ['B', 'kiB', 'MiB', 'GiB'];
+
+// Take the rate in bytes and keep dividing it by 1k until the lowest
+// value for which we have a unit is determined. Return the value with
+// up to two decimals as a string and unit/s appended.
+const displayRate = (rate) => {
+  let i = 0;
+  
+  while (rate > 1000 && i < units.length) {
+    rate /= 1000;
+    i++;
+  }
+
+  return Intl.NumberFormat(undefined, {maximumFractionDigits: 2})
+    .format(rate || 0) + ` ${units[i]}/s`;
+}
+
+export default {
+  name: 'rTorrent',
+  props: {item: Object},
+  components: {Generic},
+  // Properties for download, upload, torrent count and errors.
+  data: () => ({dl: null, ul: null, count: null, error: null}),
+  // Computed properties for the rate labels.
+  computed: {
+    downRate: function() {
+      return displayRate(this.dl);
+    },
+    upRate: function() {
+      return displayRate(this.ul);
+    },
+  },
+  created() {
+    // Set intervals if configured so the rates and/or torrent count
+    // will be updated.
+    const rateInterval = parseInt(this.item.rateInterval, 10) || 0;
+    const torrentInterval = parseInt(this.item.torrentInterval, 10) || 0;
+
+    if (rateInterval > 0) {
+      setInterval(() => this.fetchRates(), rateInterval);
+    }
+
+    if (torrentInterval > 0) {
+      setInterval(() => this.fetchCount(), torrentInterval);
+    }
+    
+    // Fetch the initial values.
+    this.fetchRates();
+    this.fetchCount();
+  },
+  methods: {
+    // Perform two calls to the XML-RPC service and fetch download
+    // and upload rates. Values are saved to the `ul` and `dl`
+    // properties.
+    fetchRates: async function() {
+      this.getRate('throttle.global_up.rate', (ul) => this.ul = ul)
+        .then(() => this.getRate('throttle.global_down.rate', (dl) => this.dl = dl))
+        .catch(() => this.error = true);
+    },
+    // Perform a call to the XML-RPC service to fetch the number of
+    // torrents.
+    fetchCount: async function() {
+      this.getCount().catch(() => this.error = true);
+    },
+    // Fetch a numeric value from the XML-RPC service by requesting
+    // the specified method name and parsing the XML. The response
+    // is expected to adhere to the structure of a single numeric
+    // value.
+    getRate: async function(methodName, callback) {
+      return this.getXml(methodName)
+        .then((xml) => {
+          const rate = parseInt(xml.getElementsByTagName('value')[0].firstChild.textContent, 10);
+
+          if (callback) {
+            callback(rate);
```

Comment:
You should directly `return rate` (so you would seemlessly get a promise with the value you need), this way, no need to mix callback & promise.